src/Npgsql/NpgsqlConnectionStringBuilder.cs

```diff
@@ -1735,4 +1802,36 @@ enum ReplicationMode
     Logical
 }
 
+/// <summary>
+/// Specifies which authentication methods are supported.
+/// </summary>
+[Flags]
+enum RequireAuthMode
+{
+    /// <summary>
+    /// Plaintext password.
+    /// </summary>
+    Password = 1,
+    /// <summary>
+    /// MD5 hashed password.
+    /// </summary>
+    MD5 = 2,
+    /// <summary>
+    /// Kerberos.
+    /// </summary>
+    GSS = 4,
+    /// <summary>
+    /// Windows SSPI.
+    /// </summary>
+    SSPI = 8,
+    /// <summary>
+    /// SASL.
+    /// </summary>
+    ScramSHA256 = 16,
+    /// <summary>
+    /// No authentication exchange.
+    /// </summary>
+    None = 32
```

Comment:
Maybe make this 0? This way, its negation means "all auth methods", which is what `!none` is defined to mean?

Reply:
I'm not sure it's going to work. For example, let's say we want either plaintext (that's 1) or none (0). 1 + 0 = 1, so we can't even say whether we allow with no authentication or not.

Reply:
You're right... None is indeed an "authentication mode" in that sense, and not the lack of one.